src/server.rs

```diff
@@ -35,33 +34,44 @@ pub async fn start_server(
 
     let mut module = RpcModule::new(());
 
+    let cached_methods = config
+        .rpcs
+        .methods
+        .iter()
+        .filter(|m| m.cache)
+        .map(|m| m.method.clone())
+        .collect::<Vec<String>>();
+
     let client = Arc::new(client);
     let _api = Api::new(client.clone());
 
-    let upstream = Arc::new(call::UpstreamMiddleware::new(client.clone()));
+    let middlewares = Arc::new(Middlewares::new(
+        vec![
+            Arc::new(CacheMiddleware::new(cached_methods, 2048)),
```

Comment:
I want each individual method should have their own instance of the middlewares so we don't need extra filter step on the middleware to decide if we want to process it or not

Reply:
make cache size configurable